okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/OkHttpClients.java

```diff
@@ -244,6 +245,12 @@ private static RemotingOkHttpClient createInternal(
 
         // cipher setup
         client.connectionSpecs(createConnectionSpecs(config.enableGcmCipherSuites()));
+        // gcm ciphers are required for http/2 per https://tools.ietf.org/html/rfc7540#section-9.2.2
```

Comment:
I thought that RFC was more prescribing the behaviour of servers - it seems like they don't define the `deployment` term in https://tools.ietf.org/html/rfc7540#page-6, but do use `endpoint` to mean `Either the client or server of the connection.`

I think I'd almost prefer to have the hard off switch for HTTP2, and just use put a warning flag log line instead?

Reply:
The rfc is describing servers, but not all servers implement it properly. This leads to confusion when we assume clients with gcm ciphers disabled can't use http/2.

Happy to put up an implementation with an http/2 flag as well, but I'd prefer if we implemented both.

Reply:
api portion of the http2 flag: https://github.com/palantir/conjure-java-runtime-api/pull/443